modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/upgrade/v3_2_0/test/FriendlyURLEntryUpgradeProcessTest.java

```diff
@@ -0,0 +1,152 @@
+import com.liferay.friendly.url.service.FriendlyURLEntryLocalService;
+import com.liferay.portal.kernel.dao.orm.EntityCacheUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Balazs Breier
+ */
+@RunWith(Arquillian.class)
+public class FriendlyURLEntryUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testUpdateFriendlyURLEntryGroupId() throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext();
+
+		long groupId = TestPropsValues.getGroupId();
+
+		FriendlyURLEntry friendlyURLEntry1 =
+			_friendlyURLEntryLocalService.addFriendlyURLEntry(
+				groupId,
+				_classNameLocalService.getClassNameId(AssetCategory.class), 0,
+				RandomTestUtil.randomString(), serviceContext);
+
+		friendlyURLEntry1.setGroupId(0);
+
+		friendlyURLEntry1 =
+			_friendlyURLEntryLocalService.updateFriendlyURLEntry(
+				friendlyURLEntry1);
+
+		FriendlyURLEntryLocalization friendlyURLEntryLocalization1 =
+			_friendlyURLEntryLocalService.getFriendlyURLEntryLocalization(
+				friendlyURLEntry1.getFriendlyURLEntryId(), "en_US");
+
+		friendlyURLEntryLocalization1.setGroupId(0);
+
+		_friendlyURLEntryLocalService.updateFriendlyURLEntryLocalization(
+			friendlyURLEntry1, "en_US", RandomTestUtil.randomString());
+
+		FriendlyURLEntry friendlyURLEntry2 =
+			_friendlyURLEntryLocalService.addFriendlyURLEntry(
+				groupId, _classNameLocalService.getClassNameId(CProduct.class),
+				0, RandomTestUtil.randomString(), serviceContext);
+
+		friendlyURLEntry2.setGroupId(0);
+
+		friendlyURLEntry2 =
+			_friendlyURLEntryLocalService.updateFriendlyURLEntry(
+				friendlyURLEntry2);
+
+		FriendlyURLEntryLocalization friendlyURLEntryLocalization2 =
+			_friendlyURLEntryLocalService.getFriendlyURLEntryLocalization(
+				friendlyURLEntry2.getFriendlyURLEntryId(), "en_US");
+
+		friendlyURLEntryLocalization2.setGroupId(0);
+
+		_friendlyURLEntryLocalService.updateFriendlyURLEntryLocalization(
+			friendlyURLEntry2, "en_US", RandomTestUtil.randomString());
```

Comment:
just a question: why do you need two "items"?

Reply:
@smottal pls check their classNameIds, they have to be different, see in the upgradeProcess.

Reply:
Technically, if one works the other should, as well, but this seemed more logical for me.